tests/integration-tests/tests/common/schedulers_common.py

```diff
@@ -329,15 +329,15 @@ def assert_job_succeeded(self, job_id, children_number=0):  # noqa: D102
         result = self._remote_command_executor.run_remote_command("scontrol show jobs -o {0}".format(job_id))
         assert_that(result.stdout).contains("JobState=COMPLETED")
 
-    def compute_nodes_count(self):  # noqa: D102
-        result = self._remote_command_executor.run_remote_command("sinfo --Node --noheader | grep compute | wc -l")
-        # split()[-1] to extract last line and trim whitespaces
-        return int(result.stdout.split()[-1])
-
-    def get_compute_nodes(self):  # noqa: D102
-        result = self._remote_command_executor.run_remote_command(
-            "sinfo --Node --noheader | grep compute | awk '{print $1}'"
-        )
+    def compute_nodes_count(self, filter_by_partition=None):  # noqa: D102
+        return len(self.get_compute_nodes(filter_by_partition))
+
+    def get_compute_nodes(self, filter_by_partition=None):  # noqa: D102
+        command = "sinfo --Node --noheader --responding"
+        if filter_by_partition:
+            command += " --partition {}".format(filter_by_partition)
+        command += " | grep -v '[~#]' | awk '{print $1}'"
```

Comment:
I would also add [*] to list of suffix to filter out, so `[~#*]`. I will address this in my changes, so ok for now